File: Completed/Go/1165.go
func calculateTime(keyboard string, word string) int {
    keyboardCost := map[rune]int{}
    for i, key := range keyboard {
        keyboardCost[key] = i
    }
    
    res := 0
    lastKey := 0
    for _, key := range word {
        newKey := keyboardCost[key]
        if newKey < lastKey {
            res += lastKey - newKey
        } else {
            res += newKey - lastKey
        }
        lastKey = newKey
    }
    return res
}

File: Completed/python3/1165.py
class Solution:
    def calculateTime(self, keyboard: str, word: str) -> int:
        keyDict = {}
        for i, key in enumerate(keyboard):
            keyDict[key]= i
            
        currIdx = 0
        time = 0
        for key in word:
            nextIdx = keyDict[key]
            time += abs(currIdx - nextIdx)
            currIdx = nextIdx
            
        return time
            
